.github/workflows/draft-release.yaml

```diff
@@ -19,28 +20,64 @@
 
 
 ---
-name: "Draft new release"
+name: "Draft Release"
+run-name: "Draft Release ${{ inputs.version }} from ${{ github.ref_name }}"
 
 on:
   workflow_dispatch:
     inputs:
       version:
-        description: 'The version you want to release.'
+        description: 'The version you want to release. Ref should be either main for latest releases or a tag for bugfixes. '
         required: true
 
 jobs:
+  validate-release:
+    name: "Validate if version can be released"
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_name == 'main' || startsWith(github.ref, 'refs/tags/') }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - id: check-tag
+        name: Check if tag exists
+        run: |-
+
+          tag=$(git tag -l ${{ inputs.version }})
+
+          if [ ! -z $tag ];
+          then
+            echo "Tag already exists! Please choose another tag."
+            exit 1
+          fi
+
+  resolve-branch-prefix:
+     name: "Resolve branch prefix (release or bugfix)"
+     needs: validate-release
+     runs-on: ubuntu-latest
+     outputs:
+       branch_name: ${{ steps.release.outputs.branch_name || steps.bugfix.outputs.branch_name }}
+     steps:
+       - id: release
```

Comment:
Personally, I prefer a simple shell script over conditional steps or dedicated actions, because shell scripts can be easily replicated in a local environment and are generally more easily understandable.

Reply:
I refactored the validation jobs in [788baa1](https://github.com/eclipse-tractusx/tractusx-edc/pull/1543/commits/788baa13d5600130017551642181721f394b7682) since I also agree those could be simpler.

On a personal level, I prefer the conditionals on the jobs or actions. The workflow execution graph is calculated before it is sent to the runner and false conditionals are skipped. When having everything in a shell script you have to be more careful what behaviors should lead to job failure and explicitly have it fail.

I'm fine either way. Let me know if the refactor is ok.

Reply:
We've seen this numerous times, where we want to replicate the runner's behaviour and debug or test a workflow locally, e.g. after a refactoring or a maintenance change. If everything is a job with conditional execution, there is zero chance of doing that, and ACT only gets you so far.

Let's get this merged as-is so as not to block the PR. 

Generally speaking, I would be very cautious when adding GH actions, because we are effectively introducing undebuggable foreign code, that you have no control over.